### ISSUE TITLE
[Feature]: Support sigmoid for classification models

### ISSUE THREAD

**Issue by tonyay163** [vllm-project/vllm#11881]
### 🚀 The feature, motivation and pitch

Right now, we only support softmax for classification models but sigmoid is equally as prevalent, depending on the usecase. I think the simplest way is just to return the logits and let us handle it ourselves. Alternatively, returning both softmax and sigmoid is probably not too much more computation.

### Additional context

See: https://wandb.ai/amanarora/Written-Reports/reports/Understanding-Logits-Sigmoid-Softmax-and-Cross-Entropy-Loss-in-Deep-Learning--Vmlldzo0NDMzNTU3#conclusion

> Thus, sigmoid is preferred for binary & multi-label classification problems whereas softmax is preferred for multi-class classification problems where we want the model to “pick a class”.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by tonyay163** [comment]
I think I found a workaround:

```
model = LLM(
    model="...",
    task="classify",
    override_pooler_config=PoolerConfig(
        pooling_type="LAST",
        normalize=False,
        softmax=False,
    )
)

def sigmoid(x: List[float]) -> List[float]:
    return [1 / (1 + math.exp(-val)) for val in x]

outputs = model.classify(text)
probs = sigmoid(outputs[0].outputs.probs)
```

**Comment by maxdebayser** [comment]
Related PRs: https://github.com/vllm-project/vllm/pull/18313 and https://github.com/vllm-project/vllm/pull/18838
